### DISCUSSION TITLE
USB Nodes

### DISCUSSION THREAD

**Discussion by Fr33man85** [RotorHazard/RotorHazard#627] [Q&A]
Hi,

we are trying to use some USB nodes in addition to our 4Node PCB board.

i want to ask, if it is possible to "dynamical" recognize the attached USB nodes.

For now it is, that the server is not starting when not all defined ttyUSB* are present.
Nor it is possible to quickly attatch some more nodes if needed (we have to ssh-ing to server to edit config.json and restart)

greetings
manuel

**Comment by ethomas997** [comment]
Yes, support for optional serial-port nodes would be a good feature.  Probably the way to do it would be via a new configuration item called OPTIONAL_SERIAL_PORTS.

**Comment by Fr33man85** [comment]
My idea was to create a "simple" script, what can be added to the startup-script of the server. 
This scripts should looking for /dev/ttyUSB* devices and moves/overrides the config.json with a prepared config.json in the server-folder corresponding to the amount of attached/found ttyUSB* devices. After that, the server starts up.

unfortunaly, iam not enough "scripter" to do this by my own without spending hours into it.
